source/server/active_tcp_socket.cc

```diff
@@ -89,14 +105,25 @@ void ActiveTcpSocket::continueFilterChain(bool success) {
           no_error = false;
           break;
         } else {
+          // There may already have data peeked due to previous filter.
+          if ((*iter_)->inspectSize() > 0 && listener_filter_buffer_->length() > 0) {
+            Network::FilterStatus status = (*iter_)->onData(*listener_filter_buffer_);
+            // The filter needn't to inspect more data, then go to next filter.
+            if (status == Network::FilterStatus::Continue) {
+              continue;
+            }
+          }
           // Blocking at the filter but no error
           return;
         }
       }
     }
     // Successfully ran all the accept filters.
     if (no_error) {
-      newConnection();
+      listener_filter_buffer_->reset();
+      if (listener_filter_buffer_->drainFromSocket()) {
+        newConnection();
+      }
```

Comment:
Seems like there needs to be error handling here?

Reply:
if `listener_filter_buffer_->drainFromSocket()` return false, then it is going to this line listener_filter_buffer_->drainFromSocket(), it will unlink and delete the ActiveTcpSocket, then everything stopped for this connection.

Reply:
Sorry you mean because it calls the close callback during the call? Can you add more comments?

Reply:
emm... I think I needn't call the close callback inside drainFromSocket(), because it is same with here. Both of them will go to the line https://github.com/envoyproxy/envoy/pull/17395/files#diff-6708370e112a5097f06e16009c3730b34f5ab0ee4f1e736f8816848409435888R140

it will unlink and delete the ActiveTcpSocket without create new connection.